azure/services/managedclusters/spec.go

```diff
@@ -327,6 +324,11 @@ func (s *ManagedClusterSpec) Parameters(existing interface{}) (params interface{
 		// AgentPool changes are managed through AMMP.
 		managedCluster.AgentPoolProfiles = existingMC.AgentPoolProfiles
 
+		// Do not trigger an update because of nil/empty discrepancies between the two sets of tags.
+		if len(existingMC.Tags) == 0 && len(managedCluster.Tags) == 0 {
+			existingMC.Tags = managedCluster.Tags
```

Comment:
would this be better?

```
existingMC.Tags, managedCluster.Tags = nil, nil
```

setting the "existing" to the "local" seems to suggest we're preferring the "local" value, when in fact what we're really trying to do is to ensure that they both have the same value so that a diff isn't computed between them

is that right?

Reply:
I suppose what I was trying to enforce here is that the resulting `managedCluster.Tags` always resolves to a non-nil, empty map when the `spec.additionalTags` are nil or empty. But I see now that it looks fine for `managedCluster.Tags` to be nil when `existingMC.Tags` is also empty since our desired state already matches the actual state in that case. Overall I think your suggestion is a bit more clear so I'll test that to make sure it works and make that change unless I run into any issues.